src/components/Dropdown.js

```diff
@@ -0,0 +1,136 @@
+import SVGIcon from "@/src/components/SVGIcon";
+import {
+  DropdownContainer,
+  DropdownLabel,
+  DropdownButton,
+  DropdownMenu,
+  DropdownItem,
+  ErrorMessage,
+  IconWrapper,
+} from "@/src/components/styles/Dropdown";
+
+const Dropdown = ({
+  label,
+  name,
+  options,
+  buttonText,
+  onOptionClick,
+  required,
+  errorMessage,
+  defaultSelected,
+}) => {
+  // State to manage the open/close state of the dropdown
+  const [isOpen, setIsOpen] = useState(false);
+  // State to manage the selected option, initialized with defaultSelected if provided
+  const [selectedOption, setSelectedOption] = useState(defaultSelected || "");
+
+  // Effect to close dropdown when clicking outside of it
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (isOpen && !event.target.closest(".dropdown-container")) {
+        setIsOpen(false);
+      }
+    };
+
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [isOpen]);
+
+  // Toggle dropdown open/close state
+  const toggleDropdown = useCallback((event) => {
+    event.preventDefault();
+    setIsOpen((prev) => !prev);
+  }, []);
+
+  // Handle option selection
+  const handleOptionClick = useCallback(
+    (option) => {
+      setSelectedOption(option);
+      setIsOpen(false);
+      onOptionClick(option);
+    },
+    [onOptionClick]
+  );
+
+  // Handle key down events for the dropdown button
+  const handleKeyDown = useCallback((event) => {
+    if (["Enter", " ", "ArrowDown"].includes(event.key)) {
+      event.preventDefault();
+      setIsOpen(true);
+      setTimeout(() => {
+        document.querySelector('[role="option"]')?.focus();
+      }, 0);
+    } else if (event.key === "Escape") {
+      setIsOpen(false);
+    }
+  }, []);
+
+  // Handle key down events for the dropdown menu
+  const handleMenuKeyDown = useCallback(
+    (event) => {
+      const currentIndex = options.findIndex(
+        (option) => option === event.target.innerText
+      );
+      if (event.key === "ArrowDown" || event.key === "ArrowUp") {
+        event.preventDefault();
+        const offset = event.key === "ArrowDown" ? 1 : -1;
+        const nextIndex =
+          (currentIndex + offset + options.length) % options.length;
+        document.getElementById(`option-${nextIndex}`)?.focus();
+      } else if (event.key === "Escape") {
+        setIsOpen(false);
+      } else if (["Enter", " "].includes(event.key)) {
+        event.preventDefault();
+        handleOptionClick(event.target.innerText);
+      }
+    },
+    [options, handleOptionClick]
+  );
+
+  return (
+    <DropdownContainer className="dropdown-container">
+      <DropdownLabel htmlFor={name}>{label}</DropdownLabel>
+      <input
+        type="hidden"
+        name={name}
+        value={selectedOption}
+        required={required}
+      />
```

Comment:
I do not know if this is intended but your input is required and hidden at the same time. 
Wouldn't that cause a problem?

Reply:
It is common to use hidden input fields in forms to transfer the value. The mandatory information should be checked before submitting the form.